mypy errors for CloudFormation custom resource types

When running `mypy src`, I see the following errors:

```plain
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "RequestType" while merging
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "RequestId" while merging
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "ResponseURL" while merging
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "ResourceType" while merging
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "LogicalResourceId" while merging
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "StackId" while merging
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "ResourceProperties" while merging
src/aws_lambda_typing/events/cloud_formation_custom_resource.py:83: error: Overwriting TypedDict field "PhysicalResourceId" while merging
Found 8 errors in 1 file (checked 34 source files)
```

This occurs because of the following in the indicated file (doc comments removed):

```python
class CloudFormationCustomResourceEvent(
    CloudFormationCustomResourceUpdate,
    CloudFormationCustomResourceDelete,
):
    pass
```

and both `CloudFormationCustomResourceUpdate` and `CloudFormationCustomResourceDelete` extend `CloudFormationCustomResourceCommon`.

I don't know the reasoning behind structuring the types this way, but perhaps a slight refactoring along the following lines would be an acceptable way to eliminate these errors, and also provide additional type safety.

First, remove `RequestType` from `CloudFormationCustomResourceCommon` so that each extension is required to specify the corresponding Literal type.

Then, define `CloudFormationCustomResourceCreate`, which is missing:

```python
class CloudFormationCustomResourceCreate(
    CloudFormationCustomResourceCommon,
    total=False,
):
    RequestType: Literal["Create"]
```

Similarly, for `CloudFormationCustomResourceUpdate` and `CloudFormationCustomResourceDelete`, set `RequestType` to `Literal["Update"]` and `Literal["Delete"]`, respectively.

Finally, define `CloudFormationCustomResourceEvent` as `Union[CloudFormationCustomResourceCreate, CloudFormationCustomResourceUpdate, CloudFormationCustomResourceDelete]` to eliminate the mypy errors above.

--- a/.github/workflows/publish.yaml
+++ b/.github/workflows/publish.yaml
@@ -37,6 +37,7 @@ jobs:
       - name: Test
         run: |
           source $(poetry env info --path)/bin/activate # activate virtual environment
+          mypy src
           mypy tests
 
       - name: Build
@@ -77,6 +78,7 @@ jobs:
       - name: Test
         run: |
           source $(poetry env info --path)/bin/activate # activate virtual environment
+          mypy src
           mypy tests
 
       - name: Build

--- a/.github/workflows/test.yaml
+++ b/.github/workflows/test.yaml
@@ -41,6 +41,7 @@ jobs:
       - name: Test
         run: |
           source $(poetry env info --path)/bin/activate # activate virtual environment
+          mypy src
           mypy tests
 
       - name: Validate Project Version

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "aws-lambda-typing"
-version = "2.9.4"
+version = "2.9.5"
 description = "A package that provides type hints for AWS Lambda event, context and response objects"
 authors = ["Mousa Zeid Baker"]
 license = "MIT"

--- a/src/aws_lambda_typing/events/__init__.py
+++ b/src/aws_lambda_typing/events/__init__.py
@@ -9,13 +9,22 @@ from .api_gateway_proxy import APIGatewayProxyEventV1 as APIGatewayProxyEventV1
 from .api_gateway_proxy import APIGatewayProxyEventV2 as APIGatewayProxyEventV2
 from .app_sync_resolver import AppSyncResolverEvent as AppSyncResolverEvent
 from .cloud_formation_custom_resource import (
+    CloudFormationCustomResourceCreate as CloudFormationCustomResourceCreate,
+)
+from .cloud_formation_custom_resource import (
+    CloudFormationCustomResourceDelete as CloudFormationCustomResourceDelete,
+)
+from .cloud_formation_custom_resource import (
     CloudFormationCustomResourceEvent as CloudFormationCustomResourceEvent,
+)
+from .cloud_formation_custom_resource import (
+    CloudFormationCustomResourceUpdate as CloudFormationCustomResourceUpdate,
 )
 from .cloud_watch_events import (
     CloudWatchEventsMessageEvent as CloudWatchEventsMessageEvent,
 )
 from .cloud_watch_logs import CloudWatchLogsEvent as CloudWatchLogsEvent
-from .code_commit import CodeCommitMessageEvent
+from .code_commit import CodeCommitMessageEvent as CodeCommitMessageEvent
 from .code_pipeline import CodePipelineEvent as CodePipelineEvent
 from .config import ConfigEvent as ConfigEvent
 from .dynamodb_stream import DynamoDBStreamEvent as DynamoDBStreamEvent

--- a/src/aws_lambda_typing/events/api_gateway_authorizer.py
+++ b/src/aws_lambda_typing/events/api_gateway_authorizer.py
@@ -5,7 +5,7 @@ import sys
 if sys.version_info >= (3, 8):
     from typing import Any, Dict, Literal, TypedDict
 else:
-    from typing import Dict
+    from typing import Any, Dict
 
     from typing_extensions import Literal, TypedDict
 

--- a/src/aws_lambda_typing/events/app_sync_resolver.py
+++ b/src/aws_lambda_typing/events/app_sync_resolver.py
@@ -5,7 +5,7 @@ import sys
 if sys.version_info >= (3, 8):
     from typing import Any, Dict, List, TypedDict, Union
 else:
-    from typing import Dict, List
+    from typing import Any, Dict, List, Union
 
     from typing_extensions import TypedDict
 

--- a/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
+++ b/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
@@ -97,3 +97,27 @@ CloudFormationCustomResourceEvent = Union[
     CloudFormationCustomResourceUpdate,
     CloudFormationCustomResourceDelete,
 ]
+"""
+CloudFormationCustomResourceEvent
+https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requesttypes.html
+
+Attributes:
+----------
+RequestType: Literal["Create", "Update", "Delete"]
+
+RequestId: str
+
+ResponseURL: str
+
+ResourceType: str
+
+LogicalResourceId: str
+
+StackId: str
+
+ResourceProperties: Dict[str, Any]
+
+PhysicalResourceId: str (only in "Update" and "Delete" events)
+
+OldResourceProperties: Dict[str, Any] (only in "Update" events)
+"""

--- a/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
+++ b/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
@@ -1,6 +1,7 @@
 #!/usr/bin/env python
 
 import sys
+from typing import Union
 
 if sys.version_info >= (3, 8):
     from typing import Any, Dict, Literal, TypedDict
@@ -17,8 +18,6 @@ class CloudFormationCustomResourceCommon(TypedDict):
 
     Attributes:
     ----------
-    RequestType: Literal["Create", "Update", "Delete"]
-
     RequestId: str
 
     ResponseURL: str
@@ -32,17 +31,28 @@ class CloudFormationCustomResourceCommon(TypedDict):
     ResourceProperties: Dict[str, Any]
     """
 
-    RequestType: Literal[
-        "Create",
-        "Update",
-        "Delete",
-    ]
     RequestId: str
     ResponseURL: str
     ResourceType: str
     LogicalResourceId: str
     StackId: str
     ResourceProperties: Dict[str, Any]
+
+
+class CloudFormationCustomResourceCreate(
+    CloudFormationCustomResourceCommon,
+    total=False,
+):
+    """
+    CloudFormationCustomResourceCreate
+    https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requesttypes-create.html
+
+    Attributes:
+    ----------
+    RequestType: Literal["Create"]
+    """
+
+    RequestType: Literal["Create"]
 
 
 class CloudFormationCustomResourceUpdate(
@@ -55,11 +65,14 @@ class CloudFormationCustomResourceUpdate(
 
     Attributes:
     ----------
+    RequestType: Literal["Update"]
+
     PhysicalResourceId: str
 
     OldResourceProperties: Dict[str, Any]
     """
 
+    RequestType: Literal["Update"]
     PhysicalResourceId: str
     OldResourceProperties: Dict[str, Any]
 
@@ -74,39 +87,17 @@ class CloudFormationCustomResourceDelete(
 
     Attributes:
     ----------
+    RequestType: Literal["Delete"]
+
     PhysicalResourceId: str
     """
 
+    RequestType: Literal["Delete"]
     PhysicalResourceId: str
 
 
-class CloudFormationCustomResourceEvent(
+CloudFormationCustomResourceEvent = Union[
+    CloudFormationCustomResourceCreate,
     CloudFormationCustomResourceUpdate,
     CloudFormationCustomResourceDelete,
-):
-    """
-    CloudFormationCustomResourceEvent
-    https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/crpg-ref-requesttypes.html
-
-    Attributes:
-    ----------
-    RequestType: Literal["Create", "Update", "Delete"]
-
-    RequestId: str
-
-    ResponseURL: str
-
-    ResourceType: str
-
-    LogicalResourceId: str
-
-    StackId: str
-
-    ResourceProperties: Dict[str, Any]
-
-    PhysicalResourceId: Optional[str] Only in Update/Delete events
-
-    OldResourceProperties: Optional[Dict[str, Any]] Only in Update events
-    """
-
-    pass
+]

--- a/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
+++ b/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
@@ -40,7 +40,6 @@ class CloudFormationCustomResourceCommon(TypedDict):
 
 class CloudFormationCustomResourceCreate(
     CloudFormationCustomResourceCommon,
-    total=False,
 ):
     """
     CloudFormationCustomResourceCreate
@@ -56,7 +55,6 @@ class CloudFormationCustomResourceCreate(
 
 class CloudFormationCustomResourceUpdate(
     CloudFormationCustomResourceCommon,
-    total=False,
 ):
     """
     CloudFormationCustomResourceUpdate
@@ -78,7 +76,6 @@ class CloudFormationCustomResourceUpdate(
 
 class CloudFormationCustomResourceDelete(
     CloudFormationCustomResourceCommon,
-    total=False,
 ):
     """
     CloudFormationCustomResourceDelete

--- a/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
+++ b/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
@@ -1,12 +1,11 @@
 #!/usr/bin/env python
 
 import sys
-from typing import Union
 
 if sys.version_info >= (3, 8):
-    from typing import Any, Dict, Literal, TypedDict
+    from typing import Any, Dict, Literal, TypedDict, Union
 else:
-    from typing import Any, Dict
+    from typing import Any, Dict, Union
 
     from typing_extensions import Literal, TypedDict
 

--- a/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
+++ b/src/aws_lambda_typing/events/cloud_formation_custom_resource.py
@@ -49,7 +49,7 @@ class CloudFormationCustomResourceCreate(
 
     Attributes:
     ----------
-    RequestType: Literal["Create"]
+    RequestType: Literal['Create']
     """
 
     RequestType: Literal["Create"]
@@ -65,7 +65,7 @@ class CloudFormationCustomResourceUpdate(
 
     Attributes:
     ----------
-    RequestType: Literal["Update"]
+    RequestType: Literal['Update']
 
     PhysicalResourceId: str
 
@@ -87,7 +87,7 @@ class CloudFormationCustomResourceDelete(
 
     Attributes:
     ----------
-    RequestType: Literal["Delete"]
+    RequestType: Literal['Delete']
 
     PhysicalResourceId: str
     """

--- a/tests/events/test_cloud_formation_custom_resource.py
+++ b/tests/events/test_cloud_formation_custom_resource.py
@@ -1,8 +1,15 @@
 from aws_lambda_typing.events import (
     CloudFormationCustomResourceCreate,
     CloudFormationCustomResourceDelete,
+    CloudFormationCustomResourceEvent,
     CloudFormationCustomResourceUpdate,
 )
+
+
+def handle_event(event: CloudFormationCustomResourceEvent) -> None:
+    if event["RequestType"] == "Update":
+        # event is automatically narrowed to CloudFormationCustomResourceUpdate
+        old_properties = event["OldResourceProperties"]
 
 
 def test_cloud_formation_custom_resource_create_event() -> None:

--- a/tests/events/test_cloud_formation_custom_resource.py
+++ b/tests/events/test_cloud_formation_custom_resource.py
@@ -6,7 +6,9 @@ from aws_lambda_typing.events import (
 )
 
 
-def handle_event(event: CloudFormationCustomResourceEvent) -> None:
+def test_handle_cloud_formation_custom_resource_event(
+    event: CloudFormationCustomResourceEvent,
+) -> None:
     if event["RequestType"] == "Update":
         # event is automatically narrowed to CloudFormationCustomResourceUpdate
         old_properties = event["OldResourceProperties"]

--- a/tests/events/test_cloud_formation_custom_resource.py
+++ b/tests/events/test_cloud_formation_custom_resource.py
@@ -1,8 +1,12 @@
-from aws_lambda_typing.events import CloudFormationCustomResourceEvent
+from aws_lambda_typing.events import (
+    CloudFormationCustomResourceCreate,
+    CloudFormationCustomResourceDelete,
+    CloudFormationCustomResourceUpdate,
+)
 
 
 def test_cloud_formation_custom_resource_create_event() -> None:
-    event: CloudFormationCustomResourceEvent = {
+    event: CloudFormationCustomResourceCreate = {
         "RequestType": "Create",
         "RequestId": "unique id for this create request",
         "ResponseURL": "pre-signed-url-for-create-response",
@@ -18,7 +22,7 @@ def test_cloud_formation_custom_resource_create_event() -> None:
 
 
 def test_cloud_formation_custom_resource_update_event() -> None:
-    event: CloudFormationCustomResourceEvent = {
+    event: CloudFormationCustomResourceUpdate = {
         "RequestType": "Update",
         "RequestId": "unique id for this update request",
         "ResponseURL": "pre-signed-url-for-update-response",
@@ -40,7 +44,7 @@ def test_cloud_formation_custom_resource_update_event() -> None:
 
 
 def test_cloud_formation_custom_resource_delete_event() -> None:
-    event: CloudFormationCustomResourceEvent = {
+    event: CloudFormationCustomResourceDelete = {
         "RequestType": "Delete",
         "RequestId": "unique id for this delete request",
         "ResponseURL": "pre-signed-url-for-delete-response",

